src/build.rs

```diff
@@ -108,6 +114,14 @@ impl Build {
             None => "debug",
         };
 
+        // confirm that target component is included in host toolchain, or add
+        // it with `rustup` otherwise.
+        toolchain::check_target_component_with_rustc_meta(
```

Comment:
This check needs to go above line 60, otherwise the check happens after trying to build the binaries, which is too late 😆

Reply:
ah, good point. I didn't actually look too closely at where we were actually building the binaries, so that's probably my fault. I'll see if I can move it to the indicated location in that case.

Reply:
@calavera  this should be resolved, I've moved the check for the target component to the right spot. I also added a note to `final_target`  that technically it could be merged, since from my impression it should be the same value in any case.